Dockerfile

```diff
@@ -0,0 +1,19 @@
+# This is dev version to make the Cargo builds fast and should re-build and run the whole
+# app everytime the container is restarted, as Rust doesn't has a proper way of hot reloading
+
+# Keeping Rust to it's latest version for having the latest Rust Dependency.
+# Once async/await is introduced, we can hard-code the version.
+FROM rust:latest
+
+WORKDIR /app/spamwatch-api
+
+COPY . /app/spamwatch-api
+
+# This will cache the library builds, which will ultimately help in
+# re-running our apps faster.
+RUN cargo build
+
+ENV RUST_BACKTRACE 1
+
+CMD ["cargo", "run"]
+
```

Comment:
Why do you have two Dockerfiles?

Reply:
Well, to separate out deployment logics for production and dev builds. Having two separate ways of deployment, helps you manage separate concerns. For eg, in production build, you can remove the source code (atleast for Rust, as you will get a compiled binary), that got copied earlier.

Also, it helps to keep different rust versions, till the dev is properly tested and can be deployed to Production.

Reply:
Also check this - https://users.rust-lang.org/t/why-does-cargo-build-not-optimise-by-default/4150
where release build can be a lot more slower to compile.

Reply:
I'm still not sure I like it... there has to be a cleaner way to do this...make an env. or something and a script?

Reply:
I can understand your concern, and keeping things DRY, but if docker commands varies (like for dev and prod), we need separate `DOCKERFILE`s, as they are supposed to be pre-built and cached in the hub, with different tags.

Not sure if env vars can be used in images, but even if they did, they will make Docker images in-consistent.

What I can think of is, Docker itself acts as an environment specific deployment strategy (just like envs). Each Environment having it's own set of deployment instructions, unless some environments have to be deployed with the exact same build, (like staging/prod can be a single Docker file).

Reply:
I suppose. Also `Dockerfile`, **_NOT_** `DOCKERFILE`